File: Python/Problem2-PatchingArray.py
class Solution:
    def minPatches(self, nums: List[int], n: int) -> int:
        maxNumGen = 0
        lenNum = len(nums)
        output = 0
        i = 0
        while(maxNumGen < n):
            if i < lenNum:
                if maxNumGen + 1 >= nums[i]:
                    maxNumGen += nums[i]
                    i += 1
                else:
                    output += 1
                    maxNumGen += maxNumGen + 1   
            else:
                output += 1
                maxNumGen += maxNumGen + 1   
 
        return output

File: Java/Problem2-PatchingArray.java
class Solution {
    public int minPatches(int[] nums, int n) {
        long maxNumGen = 0;
        int lenNum = nums.length;
        int output = 0;
        int i = 0;
        while(maxNumGen < n){
            if(i<lenNum){
                if(maxNumGen + 1 >= nums[i]){
                    maxNumGen += nums[i];
                    i++;
                }
                else{
                    output++;
                    maxNumGen += maxNumGen + 1;
                }

            }
            else{
                output++;
                maxNumGen += maxNumGen + 1;
            }
            if(maxNumGen >= n){
                break;
            }
            
        }
        return output;  
    
}

}
